configuration/purging-serverside.md

```diff
@@ -0,0 +1,208 @@
+    }
+}
+```
+
+
+**Purging is both very powerful and also very dangerous**. Read the rest of this document carefully to make sure you completely understand how to purge and the ramifications of doing so, before using purging in your project.
+
+## How purging works server-side
+
+Purging runs on the server on a configurable schedule. 
+
+It will iterate over all users to generate a list of unique roles groups that represent every user. 
+Each group will have their purged docs saved in an individual database.
+
+Then, it will iterate over all existent contacts, collecting all reports about that contact along 
+with all sms messages that the contact has sent or received. 
+This is similar to the scoping you may have encountered when configuring [tasks](./tasks.md) and [targets](./targets.md).
+
+The configured purge function runs over all combinations of purge scope (contact + reports + messages) 
+and user context (unique list of roles) to determine which docs should be purged. 
+
+The resulting list of docs to be purged is compared to the existent purged docs so that only the differences
+are saved (old purges are reverted and new purges are added). 
+
+A document is considered purged for a user if a document with the same id, prefixed by `purge`, 
+exists in the corresponding purge database. 
+The following user:
+```json
+{
+    "name": "org.couchdb.user:<your user>",
+    "roles": [
+       "district_admin",
+       "supervisor"
+    ]
+}
+```
+would get their purges from a `medic-purged-role-<role_hash>` where `role_hash` is an md5 hash of the
+user's roles.
+
+When users sync (includes initial sync), they will only download documents that are not purged for their roles. 
+
+## How purging works client-side
+
+Purging runs on the user's device at startup, before the application booting, when any of these 
+conditions are met:
+
+- The device was just setup and so initial sync just occurred.
+- The user's roles list has changed
+- It has been more than 7 days since the user has purged.
+
+Purging will not run on startup if purge is not configured, if last purge occurred recently or if 
+the user doesn't have an internet connection. 
+
+When purging runs, the device calls an API endpoint that returns batches of doc ids that have
+been purged since last time the same device has run purge.
+The system is similar to CouchDB replication, in the sense that a `checkpointer` document 
+is saved in the corresponding server-side database, that stores the `last_seq` that the device has 
+downloaded and is used to get the next batch of ids.
+After receiving a batch of ids, the device simply deletes the indicated docs locally, 
+marking them with a `purged` flag. 
+
+## Purging function configuration
+
+To enable purging, write your purging function to `purging.js` in your project root:
```

Comment:
@dianabarsan  I think we probably need to update this to match the change in medic-conf

Reply:
Oh yes, thanks :)

Reply:
Done.